### PR TITLE
Refactoring get account logic.

### PR DESCRIPTION
**Pull request by tampajohn**

### CODE REVIEW

**Comment by tampajohn** [comment]
Resolves #12

File: cli/formatting.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+func FormatNetworkBandwidth(account *eos.AccountResp, config *columnize.Config) string {
+	delegatedNet := account.TotalResources.NetWeight.Sub(account.SelfDelegatedBandwidth.NetWeight)
+
+	output := []string{
+		"net bandwidth:",
+		fmt.Sprintf("     staked:|%s|(total stake delegated from account to self)",
+			prettifyAsset(account.SelfDelegatedBandwidth.NetWeight),
+		),
+		fmt.Sprintf("     delegated:|%s|(total stake delegated to account from others)",
```
Comment (maoueh) [inline cli/formatting.go]: The CPU part is using `delegated from account` while here is using `delegated to account`. Should probably be the same in both. I just don't know which one is the right wording.

File: cli/formatting.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+)
+
+func FormatPermissions(account *eos.AccountResp, config *columnize.Config) string {
+	output := []string{
+		"permissions:",
+	}
+	for _, perm := range account.Permissions {
+		keysValues := []string{}
+		for _, key := range perm.RequiredAuth.Keys {
+			keysValues = append(keysValues, fmt.Sprintf("%d %s", key.Weight, key.PublicKey))
```
Comment (maoueh) [inline cli/formatting.go]: Seems we will need to update that at some point as it's possible to have other things than `Keys`, like `Accounts`.

Check https://eosquery.com/account/eoscanadacom compared to `eosc get account`:

```
permissions:
     active         4:
     blacklistops   1:  1 EOS7idX86zQ6M3mrzkGQ9MGHf4btSECmcTj4i8Le59ga7CpSpZYy5
     claimer        1:  1 EOS7NFuBesBKK5XHHLtzFxm7S57Eq11gUtndrsvq3Mt3XZNMTHfqc
     day2day        1:
     eosforumdapp   1:  1 EOS7YNS1swh6QWANkzGgFrjiX8E3u8WK5CK9GMAb6EzKVNZMYhCH3
     owner          4:  1 EOS5nFqz4giFz9kRZj6bByp9bgJQw7vEdrBAvMe7QWhr1xvX5YsTY, 1 EOS6EuWTngWDhzCwKoNxmrYnY87Pr8wPGYZ54ombRhQgpojqjavZX, 2 EOS7jFKsnSaA1BFdHBdhyFHHqHMdz4gcVG74WHtofeEGBiGLj4HCs, 1 EOS8CiR3Dpa1tHAY1yH8Tyf7Ue2JYRHKWAYMeo5gBHuHM2m8gKtBJ, 2 EOS8esiBart2LgGd98CwWgupqFMGtHxBV2uueesY5mHJNPRxCXCQQ
```

Comment (maoueh) [reply cli/formatting.go]: Also, would it be possible for those with multiple keys to print them one under the above:

```
owner          4:  1 EOS5nFqz4giFz9kRZj6bByp9bgJQw7vEdrBAvMe7QWhr1xvX5YsTY
                   1 EOS6EuWTngWDhzCwKoNxmrYnY87Pr8wPGYZ54ombRhQgpojqjavZX
                   2 EOS7jFKsnSaA1BFdHBdhyFHHqHMdz4gcVG74WHtofeEGBiGLj4HCs
                   1 EOS8CiR3Dpa1tHAY1yH8Tyf7Ue2JYRHKWAYMeo5gBHuHM2m8gKtBJ
                   2 EOS8esiBart2LgGd98CwWgupqFMGtHxBV2uueesY5mHJNPRxCXCQQ
```

File: cli/formatting.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+	return rightAlignColumnize(strconv.FormatFloat(value, 'f', precision, 64), unit)
+}
+
+func prettifyTime(micro int64) string {
+	value := float64(micro)
+	unit := "μs"
+	if value > 1000000*60*60 {
+		value /= float64(1000000 * 60 * 60)
+		unit = "hr"
```
Comment (maoueh) [inline cli/formatting.go]: `h` seems more common

File: cli/formatting.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	} else if value > 1000000*60 {
+		value /= float64(1000000 * 60)
+		unit = "min"
```
Comment (maoueh) [inline cli/formatting.go]: `m` seems more common

File: cli/formatting.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+	} else if value > 1000000 {
+		value /= float64(1000000)
+		unit = "sec"
```
Comment (maoueh) [inline cli/formatting.go]: `s` for seconds.

File: cli/formatting.go [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+		fmt.Sprintf("     unstaking:|%s", prettifyAsset(totalUnstaking)),
+		fmt.Sprintf("     total:|%s", prettifyAsset(total)),
+	}
+
+	return columnize.Format(output, config)
+}
+
+func FormatProducers(account *eos.AccountResp, config *columnize.Config) string {
+	output := []string{
+		"producers:",
```
Comment (maoueh) [inline cli/formatting.go]: I think it would be clearer with `votes:` or `voted for:`, what do you think?

File: cli/formatting.go [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+	for _, name := range accounts {
+		names = append(names, string(name))
+	}
+
+	return strings.Join(names, "|  ")
+}
+
+func rightAlignColumnize(value, unit string) string {
+	w := new(tabwriter.Writer)
+	bs := bytes.NewBuffer([]byte(""))
```
Comment (maoueh) [inline cli/formatting.go]: Could have been `[]byte{}`

File: cli/formatting.go (hunk 1 above)
Comment (tampajohn) [reply cli/formatting.go]: I'll pick `to account` and update.

File: cli/formatting.go (hunk 2 above)
Comment (tampajohn) [reply cli/formatting.go]: Makes sense, can do that as well :)

File: cli/formatting.go (hunk 3 above)
Comment (tampajohn) [reply cli/formatting.go]: I chose to use the cleos unit per the original issue discussion, completely agree if we want to diverge from their units.  @abourget had originally suggested matching their units

File: cli/formatting.go (hunk 4 above)
Comment (tampajohn) [reply cli/formatting.go]: I chose to use the cleos unit per the original issue discussion, completely agree if we want to diverge from their units. @abourget had originally suggested matching their units

File: cli/formatting.go (hunk 5 above)
Comment (tampajohn) [reply cli/formatting.go]: I chose to use the cleos unit per the original issue discussion, completely agree if we want to diverge from their units. @abourget had originally suggested matching their units

File: cli/formatting.go (hunk 6 above)
Comment (tampajohn) [reply cli/formatting.go]: I agree, `voted for:` reads better

File: cli/formatting.go (hunk 7 above)
Comment (tampajohn) [reply cli/formatting.go]: Yeah... not sure why I did that, will fix

File: cli/formatting.go (hunk 4 above)
Comment (maoueh) [reply cli/formatting.go]: Perfect, let me check with him.

File: cli/formatting.go (hunk 2 above)
Comment (tampajohn) [reply cli/formatting.go]: Do you want some sort of character/symbol to represent waits/accounts/keys?  I mean, it's pretty obvious without them

File: cli/formatting.go (hunk 4 above)
Comment (maoueh) [reply cli/formatting.go]: We will go with standard units: `s -> seconds`, `m -> minutes` and `h -> hours`.

File: cli/formatting.go (hunk 2 above)
Comment (abourget) [reply cli/formatting.go]: @tampajohn it seems you dropped the `formatNestedPermissions` feature.. have you sync'd up with the latest code? I tweaked the permissions printout quite a bit..

Comment (tampajohn) [reply cli/formatting.go]: @abourget seems that I did when I was collapsing some of the merge commits, which explains the verbose commit history :).  I've added it back in and made it output a single line per value as @maoueh requested.

File: cli/formatting.go (hunk 1 above)
Comment (tampajohn) [reply cli/formatting.go]: Updated

File: cli/formatting.go (hunk 6 above)
Comment (tampajohn) [reply cli/formatting.go]: Updated

File: cli/formatting.go (hunk 7 above)
Comment (tampajohn) [reply cli/formatting.go]: Updated

**Comment by tampajohn** [comment]
@maoueh I think all of your requests have been addressed, please review :)

**Comment by fproulx-eoscanada** [comment]
@tampajohn Hi John. 
It's really cool to see community contribution, thank you !

I would appreciate if you'd consider merging your refactoring of the CPU/NET/RAM stuff with the changes we've made in the meantime to permissions.

Your branch output:
```
$ eosc get account eoscanadacom
permissions:
     active         4:
     blacklistops   1:  1 EOS7idX86zQ6M3mrzkGQ9MGHf4btSECmcTj4i8Le59ga7CpSpZYy5
     claimer        1:  1 EOS7NFuBesBKK5XHHLtzFxm7S57Eq11gUtndrsvq3Mt3XZNMTHfqc
     day2day        1:
     eosforumdapp   1:  1 EOS7YNS1swh6QWANkzGgFrjiX8E3u8WK5CK9GMAb6EzKVNZMYhCH3
     owner          4:  1 EOS5nFqz4giFz9kRZj6bByp9bgJQw7vEdrBAvMe7QWhr1xvX5YsTY, 1 EOS6EuWTngWDhzCwKoNxmrYnY87Pr8wPGYZ54ombRhQgpojqjavZX, 2 EOS7jFKsnSaA1BFdHBdhyFHHqHMdz4gcVG74WHtofeEGBiGLj4HCs, 1 EOS8CiR3Dpa1tHAY1yH8Tyf7Ue2JYRHKWAYMeo5gBHuHM2m8gKtBJ, 2 EOS8esiBart2LgGd98CwWgupqFMGtHxBV2uueesY5mHJNPRxCXCQQ
```

Master branch output:
```
$ eosc get account eoscanadacom
permissions:
     "owner" w/ 4                     :  +1 EOS5nFqz4giFz9kRZj6bByp9bgJQw7vEdrBAvMe7QWhr1xvX5YsTY, +1 EOS6EuWTngWDhzCwKoNxmrYnY87Pr8wPGYZ54ombRhQgpojqjavZX, +2 EOS7jFKsnSaA1BFdHBdhyFHHqHMdz4gcVG74WHtofeEGBiGLj4HCs, +1 EOS8CiR3Dpa1tHAY1yH8Tyf7Ue2JYRHKWAYMeo5gBHuHM2m8gKtBJ, +2 EOS8esiBart2LgGd98CwWgupqFMGtHxBV2uueesY5mHJNPRxCXCQQ
           "active" w/ 4              :  +2 eoscanadaaaa@active, +2 eoscanadaaab@active, +2 eoscanadaaac@active, +2 eoscanadaaad@active, +2 eoscanadaaae@active, +1 eoscanadaaaf@active, +1 eoscanadaaag@active, +1 eoscanadaaah@active, +1 eoscanadaaai@active, +1 wait 10800 seconds
                 "blacklistops" w/ 1  :  +1 EOS7idX86zQ6M3mrzkGQ9MGHf4btSECmcTj4i8Le59ga7CpSpZYy5
                 "claimer" w/ 1       :  +1 EOS7NFuBesBKK5XHHLtzFxm7S57Eq11gUtndrsvq3Mt3XZNMTHfqc
                 "day2day" w/ 1       :  +1 eoscanadaaaa@active, +1 eoscanadaaac@active, +1 eoscanadaaaf@active, +1 eoscanadaaag@active, +1 eoscanadaaah@active, +1 eoscanadaaai@active
                 "eosforumdapp" w/ 1  :  +1 EOS7YNS1swh6QWANkzGgFrjiX8E3u8WK5CK9GMAb6EzKVNZMYhCH3
```

I wonder if we could do a better printout, but I certainly prefer the one that is nested visually with tab indentation

**Comment by maoueh** [comment]
@tampajohn Perfect, I checked the permissions rendering on your current branch and it's perfect!

Merging, thanks again for your contribution(s)!

**Pull request merged**